wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -95,10 +95,9 @@ class WireGenerator(
     val errorCollector = ErrorCollector()
     val linker = Linker(loader, errorCollector, permitPackageCycles = true, loadExhaustively = true)
 
+    val sourcePaths = setOf(*request.fileToGenerateList.toTypedArray())
     val protoFiles = mutableListOf<ProtoFile>()
-    val sourcePaths = mutableSetOf<String>()
     for (fileDescriptorProto in request.protoFileList) {
```

Comment:
This list includes the files we are compiled _as well as_ all of their transitive imports. We need them all in the schema in order for Wire to be able to link everything correctly and resolve references to imported symbols. But we don't want to generate code for all of them.

This corresponds to [this field](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/compiler/plugin.proto#L74-L88) in the request.